File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        int i = 0, j = 0;
        while(i < str1.size() && j < str2.size()){
            if(str1[i] == str2[j]){
                i++;
                j++;
            }
            else{
                str1[i] = (str1[i] == 'z')?'a':(str1[i]+1);
                if(str1[i] == str2[j]){
                    i++;
                    j++;
                }
                else{
                    i++;
                }
            }
        }
        return (j >= str2.size());
    }
};

File: 3018-make-string-a-subsequence-using-cyclic-increments/make-string-a-subsequence-using-cyclic-increments.go
func canMakeSubsequence(str1 string, str2 string) bool {
    var j int = 0
    for i:=0; i < len(str1) && j < len(str2);i++{
        if str1[i] == str2[j]{
            j++
        }else{
            var ch rune = 'a'
            if str1[i] != 'z' {
                ch = rune(str1[i]+1)
            }
            if ch == rune(str2[j]){
                j++
            }
        }
    }
    return j >= len(str2)
}